### PR TITLE
[WIP] cairocffi: 0.7.2 -> 0.8.0

### PR DESCRIPTION
**Pull request by vrthra**
###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by vrthra** [comment]
The ffi fails on mac because it tries to load `libcairo.so.2` (hard coded) while on mac the library is named `libcairo.2.dynlib`.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -3468,11 +3463,16 @@ in {
       substituteInPlace cairocffi/__init__.py --subst-var-by cairo ${pkgs.cairo.out}
       substituteInPlace cairocffi/__init__.py --subst-var-by glib ${pkgs.glib.out}
       substituteInPlace cairocffi/__init__.py --subst-var-by gdk_pixbuf ${pkgs.gdk_pixbuf.out}
-    '';
+    '' + optionalString (stdenv.isDarwin) ''
```
Comment (vrthra) [inline pkgs/top-level/python-packages.nix]: I would like to know if there is a better way to do this.

**Comment by FRidh** [comment]
I completely forgot about this PR. What is the status of it?

**Comment by vrthra** [comment]
Sorry, wasn't able to focus on this yet.

**Comment by jtojnar** [comment]
Updated in https://github.com/NixOS/nixpkgs/pull/35527.

**Pull request closed without merging**